src/parser/GenericParser.ts

```diff
@@ -42,69 +42,49 @@ export class GenericParser {
     async calculateMetrics() {
         const fileMetrics = new Map<string, FileMetricResults>();
         const unsupportedFiles: string[] = [];
+        const errorFiles: string[] = [];
         let couplingMetrics = {} as CouplingResult;
 
         try {
             const filePathGenerator = findFilesAsync(this.config);
-            const parsePromises = new Map<string, Promise<SourceFile | null>>();
+            const parsePromises = new Map<string, Promise<SourceFile>>();
 
             for await (const filePath of filePathGenerator) {
-                parsePromises.set(
-                    filePath,
-                    TreeParser.parse(filePath, this.config).catch((reason) => {
-                        console.error("Error while parsing a tree for the file " + filePath);
-                        console.error(reason);
-                        return null;
-                    }),
-                );
+                parsePromises.set(filePath, TreeParser.parse(filePath, this.config));
             }
 
-            const fileMetricPromises: Promise<[string, FileMetricResults]>[] = [];
+            const fileMetricPromises: Promise<[SourceFile, FileMetricResults]>[] = [];
 
             if (this.config.parseMetrics) {
-                const metricsParser = new MetricCalculator(this.config);
+                const metricsParser = new MetricCalculator();
 
                 for (const [filePath, parsePromise] of parsePromises) {
```

Comment:
filePath is not needed anymore
```suggestion
                for (const parsePromise of parsePromises) {
```